buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/transform/feature/FeatureInjector.java

```diff
@@ -0,0 +1,133 @@
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.elasticsearch.gradle.test.rest.transform.RestTestTransformGlobalSetup;
+import org.elasticsearch.gradle.test.rest.transform.RestTestTransformGlobalTeardown;
+
+import javax.annotation.Nullable;
+import java.util.Iterator;
+
+/**
+ * A parent class for transformations that are backed by a feature. This will inject the necessary "feature" into the
+ * global setup and teardown section. See also org.elasticsearch.test.rest.yaml.Features for a list of possible features.
+ */
+public abstract class FeatureInjector implements RestTestTransformGlobalSetup, RestTestTransformGlobalTeardown {
+
+    private static JsonNodeFactory jsonNodeFactory = JsonNodeFactory.withExactBigDecimals(false);
+
+    @Override
+    public ObjectNode transformSetup(ObjectNode setupNodeParent) {
+        // check to ensure that headers feature does not already exist
+        if (setupNodeParent != null) {
+            ArrayNode setupNode = (ArrayNode) setupNodeParent.get("setup");
+            if (hasHeadersFeature(setupNode)) {
+                return setupNodeParent;
+            }
+        }
+        // transform or insert the headers feature into setup/skip/features
+        ArrayNode setupNode;
+        if (setupNodeParent == null) {
+            setupNodeParent = new ObjectNode(jsonNodeFactory);
+            setupNode = new ArrayNode(jsonNodeFactory);
+            setupNodeParent.set("setup", setupNode);
+        }
+        setupNode = (ArrayNode) setupNodeParent.get("setup");
+        addSkip(setupNode);
+        return setupNodeParent;
+    }
+
+    @Override
+    public ObjectNode transformTeardown(@Nullable ObjectNode teardownNodeParent) {
+        if (teardownNodeParent != null) {
+            ArrayNode teardownNode = (ArrayNode) teardownNodeParent.get("teardown");
+            // only transform an existing teardown section since a teardown does not inherit from setup but still needs the skip section
+            if (teardownNode != null) {
+                // check to ensure that headers feature does not already exist
+                if (hasHeadersFeature(teardownNode)) {
+                    return teardownNodeParent;
+                }
+                addSkip(teardownNode);
+                return teardownNodeParent;
+            }
+        }
+        return teardownNodeParent;
+    }
+
+    /**
+     * The name of the feature to skip. These are defined in org.elasticsearch.test.rest.yaml.Features and found in the tests at
+     * as the value of skip.feature. For example this method should return "allowed_warnings" :
+     * <pre>
+     * skip:
+     *       features: allowed_warnings
+     * </pre>
+     */
+    public abstract String getSkipFeatureName();
+
+    private boolean hasHeadersFeature(ArrayNode skipParent) {
```

Comment:
should this method now have a more generic method? I suppose, it can check for any feature not only allowed warnings?

Reply:
yup it should be generic, thanks ! fixed on [`bc7b2aa` (#68944)](https://github.com/elastic/elasticsearch/pull/68944/commits/bc7b2aa46fef51085bb1fd0293cb05be5a4b5ba4)